library/std/src/fs.rs

```diff
@@ -1676,9 +1676,9 @@ pub fn rename<P: AsRef<Path>, Q: AsRef<Path>>(from: P, to: Q) -> io::Result<()>
 /// This function will return an error in the following situations, but is not
 /// limited to just these cases:
 ///
-/// * The `from` path is not a file.
-/// * The `from` file does not exist.
-/// * The current process does not have the permission rights to access
+/// * `from` is not a regular file.
```

Comment:
Can't you copy symlinks as well?

Reply:
I am not quite sure about the behavior if it includes symlinks, does it?

Reply:
It opens the file and then checks the metadata. Symlinks are resolved during open. So symlinks are possible but must point to a regular file.

Reply:
```suggestion
/// * `from` is not a regular file or symlink to a regular file.
```
Is this good enough?

Reply:
I'd use `neither ... nor ...`

Reply:
```suggestion
/// * `from` is neither a regular file nor a symlink to a regular file.
```

Reply:
The suggested wording makes sense to me. While updating it, I think we can also improve the error message. The current is "the source path is not an existing regular file", it doesn't mention about the symlink as well.

Reply:
I believe improving the error message can be opened in another issue.

Reply:
Could you clarify why do you think so? It's user-facing and can confuse as same as the docs.

Reply:
Because this pull request is docs only, if I want to I will make another pull request to improve the error message, but I may not have time to do that now, I have quite a few tasks in my TODO.

Reply:
The goal of the PR is to make `fs::copy`'s behavior more obvious and we don't have to stick the docs. But due to your personal issue, we cannot address the error message, then it's totally fine ^^

Reply:
@pickfire Could you update the wording like https://github.com/rust-lang/rust/pull/79399#discussion_r531356543 as one commit? r=me after that.